tests/test_arrow_dataset.py

```diff
@@ -1222,6 +1222,52 @@ def test_export(self, in_memory):
             self.assertEqual(i, len(formatted_dset) - 1)
             del dset, formatted_dset
 
+    def test_to_dict(self, in_memory):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # Batched
+            dset = self._create_dummy_dataset(in_memory, tmp_dir, multiple_columns=True)
+            bacth_size = dset.num_rows - 1
+            to_dict_generator = dset.to_dict(batched=True, batch_size=bacth_size)
+
+            for batch in to_dict_generator:
+                self.assertIsInstance(batch, dict)
+                self.assertListEqual(sorted(batch.keys()), sorted(dset.column_names))
+                for col_name in dset.column_names:
+                    self.assertIsInstance(batch[col_name], list)
+                    self.assertLessEqual(len(batch[col_name]), bacth_size)
+
+            # Full
+            dset_to_dict = dset.to_dict()
+            self.assertIsInstance(dset_to_dict, dict)
+            self.assertListEqual(sorted(dset_to_dict.keys()), sorted(dset.column_names))
+            for col_name in dset.column_names:
+                self.assertIsInstance(dset_to_dict[col_name], list)
+                self.assertLessEqual(len(dset_to_dict[col_name]), dset.num_rows)
+
```

Comment:
feel free to add a test where there is an indices mapping
```
dset_with_indices_mapping = dset.select([1, 0])
```
for example.
The new `dset_with_indices_mapping` dataset is of size 2 and its first element is at the position 1 in the pyarrow table, and the second element is at the position 0 in the pyrrow table.